### PR TITLE
refactor: global options location

### PR DESCRIPTION
**Pull request by BioPhoton**
This PR could go before #246 as there are a couple of questions it would resolve.

**It includes:**
- relocating of global options
- separation responsibilities between `cli` and `core`
- type improvements for commands
- deleting empty file 👻
- small type tweaks 

**Out of scope 😇:**   
- 🧹 utils - merge file utils into 1 file

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # See http://help.github.com/ignore-files/ for more about ignoring files.
 
 .env
-
+.nx
```
Comment (matejchalk) [inline .gitignore:4]: What's in `.nx`? Ideally we should have a comment explaing why it's ignored.

File: packages/cli/src/lib/autorun/command-object.ts [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+  CollectOptions,
+  UploadOptions,
+  collectAndPersistReports,
+  upload,
+} from '@code-pushup/core';
 import { CLI_NAME } from '../cli';
-import { GeneralCliOptions } from '../implementation/model';
 
+type AutorunOptions = CollectOptions & UploadOptions;
 export function yargsAutorunCommandObject() {
```
Comment (matejchalk) [inline packages/cli/src/lib/autorun/command-object.ts:12]:
```suggestion
type AutorunOptions = CollectOptions & UploadOptions;

export function yargsAutorunCommandObject() {
```

File: packages/core/package.json [hunk 3]
```diff
@@ -5,7 +5,8 @@
     "@code-pushup/models": "*",
     "@code-pushup/utils": "*",
     "@code-pushup/portal-client": "^0.1.2",
-    "chalk": "^5.3.0"
+    "chalk": "^5.3.0",
+    "zod": "^3.22.4"
```
Comment (matejchalk) [inline packages/core/package.json:9]: Why? Only `models` should use `zod` directly, I believe. 

You don't need `globalOptionsSchema`, just the `GlobalOptions` type. Yargs doesn't use Zod schemas.

File: packages/core/src/lib/implementation/read-code-pushup-config.ts [hunk 4]
```diff
@@ -9,15 +13,16 @@ export class ConfigPathError extends Error {
 }
 
 export async function readCodePushupConfig(filepath: string) {
-  const isFile = (await stat(filepath)).isFile();
+  const validFilepath = filePathSchema('config file path').parse(filepath);
```
Comment (matejchalk) [inline packages/core/src/lib/implementation/read-code-pushup-config.ts:16]: I would revert these changes. We don't need to include Zod just for this, the "`filePathSchema`" only checks string length. Just check if the file exists with `stat`, we don't need to complicate things.

**Review by matejchalk — CHANGES_REQUESTED** [review]

File: .gitignore (hunk 1 above)
Comment (BioPhoton) [reply .gitignore:4]: done

File: packages/core/src/lib/implementation/read-code-pushup-config.ts (hunk 4 above)
Comment (BioPhoton) [reply packages/core/src/lib/implementation/read-code-pushup-config.ts:16]: done

File: packages/core/package.json (hunk 3 above)
Comment (BioPhoton) [reply packages/core/package.json:9]: ATM zod is used to provide default values. Ill refactor to yargs and types only

File: packages/core/src/lib/implementation/read-code-pushup-config.ts [hunk 5]
```diff
@@ -9,15 +9,18 @@ export class ConfigPathError extends Error {
...
 
+  const isFile = (await stat(filepath)).isFile();
   if (!isFile) {
     throw new ConfigPathError(filepath);
   }
 
   return importEsmModule<CoreConfig>(
     {
-      filepath,
+      filepath: filepath,
```
Comment (BioPhoton) [inline packages/core/src/lib/implementation/read-code-pushup-config.ts:23]:
```suggestion
      filepath,
```

**Review by matejchalk — APPROVED** [review]
LGTM :slightly_smiling_face:
